packages/x-components/src/x-modules/queries-preview/components/query-preview.vue

```diff
@@ -197,7 +198,11 @@
     protected created(): void {
       this.$watch(
         () => this.queryPreviewRequest,
-        request => this.emitQueryPreviewRequestUpdated(request)
+        (newRequest, oldRequest) => {
+          if (JSON.stringify(newRequest) !== JSON.stringify(oldRequest)) {
```

Comment:
Changing the prop and watching a computed property creates false positives when pairing this component with the semantic queries. This prevents that. (Thinking of creating a better function that compares objects and adding it to the objects utils in x-utils)

Reply:
Doesn't work for deeply nested objects, I've [opened a task](https://searchbroker.atlassian.net/browse/EMP-2123) to have a deep object comparator and then [another](https://searchbroker.atlassian.net/browse/EMP-2124) to use it here.